BitFaster.Caching/Duration.cs

```diff
@@ -65,9 +80,16 @@ public TimeSpan ToTimeSpan()
         /// <returns>A duration.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Duration FromTimeSpan(TimeSpan timeSpan)
-        {
+        {
 #if NETCOREAPP3_0_OR_GREATER
-            return new Duration((long)timeSpan.TotalMilliseconds);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
Currently, MacOS is only tested on .NET6 which uses the other code path. 

The failing test depends on this method - StopwatchTickConverter.ToTicks may be broken in this case.